framework/src/problems/EigenProblem.C

```diff
@@ -294,3 +294,27 @@ EigenProblem::isNonlinearEigenvalueSolver()
          solverParams()._eigen_solve_type == Moose::EST_MONOLITH_NEWTON ||
          solverParams()._eigen_solve_type == Moose::EST_MF_MONOLITH_NEWTON;
 }
+
+void
+EigenProblem::scaleEigenvector(const Real scaling_factor)
+{
+// FIXME: This needs to be protected since NonlinearEigenSystem derives from ParallelObject (instead
+// of NonlinearSystemBase) when SLEPc isn't installed
+#if LIBMESH_HAVE_SLEPC
+  std::vector<VariableName> var_names = getVariableNames();
```

Comment:
Are you scaling all variables? or just eigen variables? From my understanding, we should scale only Eigen variables

Reply:
Only variables marked eigen. You can see the if statement in the following line.